src/metabase/models/params/chain_filter.clj

```diff
@@ -193,14 +193,20 @@
   (let [rows (mdb.query/query {:select    [[:fk-field.id :f1]
                                            [:fk-table.id :t1]
                                            [:pk-field.id :f2]
-                                           [:pk-field.table_id :t2]]
+                                           [:pk-field.table_id :t2]
+                                           :fk-field.active]
                                :from      [[:metabase_field :fk-field]]
-                               :left-join [[:metabase_table :fk-table]    [:= :fk-field.table_id :fk-table.id]
+                               :left-join [[:metabase_table :fk-table]    [:and [:= :fk-field.table_id :fk-table.id]
+                                                                                :fk-table.active]
                                            [:metabase_database :database] [:= :fk-table.db_id :database.id]
-                                           [:metabase_field :pk-field]    [:= :fk-field.fk_target_field_id :pk-field.id]]
+                                           [:metabase_field :pk-field]    [:and [:= :fk-field.fk_target_field_id :pk-field.id]
+                                                                                :pk-field.active]]
                                :where     [:and
                                            [:= :database.id database-id]
-                                           [:not= :fk-field.fk_target_field_id nil]]})]
+                                           [:not= :fk-field.fk_target_field_id nil]
+                                           :fk-field.active]
```

Comment:
I'm probably missing some context - why are you not checking that `pk-field` is active too?

Reply:
Ah, you're assuming an active field's `fk_target_field_id` will always point to another active field. That's unfortunately not true because of this other bug [If a foreign key constraint is dropped, sync ignores it](https://github.com/metabase/metabase/issues/39687). 

If an active field `fk_target_field_id` targets another primary key field that is dropped or renamed, the primary key field will become inactive and `fk_target_field_id` will still point to it.

Reply:
Okay you are checking for `pk-field.active`, just need to test it then